### ISSUE TITLE
"Delegation token must exist for this connector." when using pyspark

### ISSUE THREAD

**Issue by cccs-eric** [Azure/azure-event-hubs-for-kafka#102]
Description
===========
I am writing a Jupyter Notebook to read/write data to a Event Hub (from a dedicated cluster) using pyspark.  I am using SAS authentication and I was able to write some messages into my hub with the following code:

```
TOPIC = 'test'
BOOTSTRAP_SERVERS = "mynamespace.servicebus.windows.net:9093"
EH_SASL = "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"$ConnectionString\" password=\"Endpoint=sb://mynamespace.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=<key value redacted>\";"

columns = ['key', 'value']
values = [('my key', 'a value')]
df = spark.createDataFrame(values, columns)
df.write \
    .format("kafka") \
    .option("topic", TOPIC) \
    .option("kafka.bootstrap.servers", BOOTSTRAP_SERVERS) \
    .option("kafka.sasl.mechanism", "PLAIN") \
    .option("kafka.security.protocol", "SASL_SSL") \
    .option("kafka.sasl.jaas.config", EH_SASL) \
    .option("checkpointLocation", "./checkpoint") \
    .save()
```
This works since I can see the number of messages going up in the Event Hub metrics.

Now I am trying to read those messages using this code:
```
readdf = spark.read \
    .format("kafka") \
    .option("subscribe", TOPIC) \
    .option("kafka.bootstrap.servers", BOOTSTRAP_SERVERS) \
    .option("kafka.sasl.mechanism", "PLAIN") \
    .option("kafka.security.protocol", "SASL_SSL") \
    .option("kafka.sasl.jaas.config", EH_SASL) \
    .option("kafka.request.timeout.ms", "60000") \
    .option("kafka.session.timeout.ms", "60000") \
    .option("startingOffsets", "earliest") \
    .option("kafka.group.id", "myconsumergroup") \
    .option("failOnDataLoss", "false") \
    .load()

readdf.show()
```
And I get the following stacktrace:
```
Py4JJavaError: An error occurred while calling o279.showString.
: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 3.0 failed 4 times, most recent failure: Lost task 0.3 in stage 3.0 (TID 14, 10.244.10.63, executor 0): java.lang.IllegalArgumentException: requirement failed: Delegation token must exist for this connector.
	at scala.Predef$.require(Predef.scala:281)
	at org.apache.spark.kafka010.KafkaTokenUtil$.isConnectorUsingCurrentToken(KafkaTokenUtil.scala:299)
	at org.apache.spark.sql.kafka010.KafkaDataConsumer.getOrRetrieveConsumer(KafkaDataConsumer.scala:533)
	at org.apache.spark.sql.kafka010.KafkaDataConsumer.$anonfun$getAvailableOffsetRange$1(KafkaDataConsumer.scala:333)
	at org.apache.spark.util.UninterruptibleThread.runUninterruptibly(UninterruptibleThread.scala:77)
	at org.apache.spark.sql.kafka010.KafkaDataConsumer.runUninterruptiblyIfPossible(KafkaDataConsumer.scala:588)
	at org.apache.spark.sql.kafka010.KafkaDataConsumer.getAvailableOffsetRange(KafkaDataConsumer.scala:332)
	at org.apache.spark.sql.kafka010.KafkaSourceRDD.resolveRange(KafkaSourceRDD.scala:139)
	at org.apache.spark.sql.kafka010.KafkaSourceRDD.compute(KafkaSourceRDD.scala:91)
```

I have added the 'myconsumergroup' as a new Consumer Group under the 'test' Event Hub (Topic).  Why am I getting this error, and what is different between a producer and a consumer in terms of authentication?  I am using the same variables for the tokens, so it's not a copy-paste error in my code.

How to reproduce
================
This is pretty much my entire notebook, only missing the spark init code:
```
import os
from os import environ
from pyspark import SparkContext, SparkConf
from pyspark.sql import SparkSession
from pyspark.sql.types import *

masterUri = "spark://<my cluster>:7077"
spark = ( SparkSession.builder
    .master(masterUri)
    .appName("Event Hub Test")
    .config("spark.executor.memory", "2g")
    .config("spark.cores.max", 1)
    .enableHiveSupport()
    .getOrCreate())
```

Has it worked previously?
=========================
This is my first attempt at making this work with my dedicated cluster (using a private endpoint).

Other info
=========
Libraries used: 

- spark-sql-kafka-0-10_2.12-3.0.0-preview2.jar
- spark-streaming-kafka-0-10_2.12-3.0.0-preview2.jar
- spark-token-provider-kafka-0-10_2.12-3.0.0-preview2.jar
- kafka-clients-2.3.1.jar

**Comment by cccs-eric** [comment]
It seems to be a bug in Spark 3.0.0 preview2 (version I am using): https://issues.apache.org/jira/browse/SPARK-30495
I will upgrade and test.  I'll close the issue if it fixes it (most likely)

**Comment by arerlend** [comment]
@cccs-eric any update?

**Comment by cccs-eric** [comment]
Hi @arerlend , sorry I should have done a followup.  It was indeed a bug in spark and it is fixed as of Spark 3.0.0-RC1 (I built it and confirmed the fix).
